bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,536 @@
+                            } else if ((dpval[i].value) == 10) {
+                                updateState(CHANNEL_TYPE_VANESUD, StringType.valueOf("SWING"));
+                            } else if ((dpval[i].value) == 11) {
+                                updateState(CHANNEL_TYPE_VANESUD, StringType.valueOf("SWIRL"));
+                            } else if ((dpval[i].value) == 12) {
+                                updateState(CHANNEL_TYPE_VANESUD, StringType.valueOf("WIDE"));
+                            } else {
+                                updateState(CHANNEL_TYPE_VANESUD, StringType.valueOf(String.valueOf(dpval[i].value)));
+                            }
+                            break;
+                        case 6:
+                            if ((dpval[i].value) == 0) {
+                                updateState(CHANNEL_TYPE_VANESLR, StringType.valueOf("AUTO"));
+                            } else if ((dpval[i].value) == 10) {
+                                updateState(CHANNEL_TYPE_VANESLR, StringType.valueOf("SWING"));
+                            } else if ((dpval[i].value) == 11) {
+                                updateState(CHANNEL_TYPE_VANESLR, StringType.valueOf("SWIRL"));
+                            } else if ((dpval[i].value) == 12) {
+                                updateState(CHANNEL_TYPE_VANESLR, StringType.valueOf("WIDE"));
+                            } else {
+                                updateState(CHANNEL_TYPE_VANESLR, StringType.valueOf(String.valueOf(dpval[i].value)));
+                            }
+                            break;
+                        case 9:
+                            int unit = Math.round((dpval[i].value) / 10);
+                            State stateValue = QuantityType.valueOf(unit, SIUnits.CELSIUS);
+                            updateState(CHANNEL_TYPE_TARGETTEMP, stateValue);
+                            break;
+                        case 10:
+                            unit = Math.round((dpval[i].value) / 10);
+                            stateValue = QuantityType.valueOf(unit, SIUnits.CELSIUS);
+                            updateState(CHANNEL_TYPE_AMBIENTTEMP, stateValue);
+                            break;
+                        case 37:
+                            unit = Math.round((dpval[i].value) / 10);
+                            stateValue = QuantityType.valueOf(unit, SIUnits.CELSIUS);
+                            updateState(CHANNEL_TYPE_OUTDOORTEMP, stateValue);
+                            break;
+                    }
+                }
+            }
+        }
+    }
+
+    public void login() {
+        String contentString = "{\"command\":\"login\",\"data\":{\"username\":\"Admin\",\"password\":\"" + password
+                + "\"}}";
+        String response = api.postRequest(ipAddress, contentString);
+        if (response != null) {
+            Response resp = gson.fromJson(response, Response.class);
```

Comment:
You should catch any `JsonSyntaxException` that might be thrown when using gson. Sometimes vendors accidentally break their own api. You should make sure that your code can appropriately handle when that happens.

Reply:
Done